expr/agg/collect.go

```diff
@@ -1,30 +1,31 @@
 package agg
 
 import (
-	"errors"
+	"fmt"
 
 	"github.com/brimdata/zed"
 	"github.com/brimdata/zed/zcode"
+	"github.com/brimdata/zed/zson"
 )
 
 type Collect struct {
 	values []zed.Value
 	size   int
 }
 
-func (c *Collect) Consume(v zed.Value) error {
-	if v.IsNil() {
-		return nil
+var _ Function = (*Collect)(nil)
+
+func (c *Collect) Consume(val *zed.Value) {
+	if !val.IsNull() {
+		c.update(val)
 	}
-	c.update(v)
-	return nil
 }
 
-func (c *Collect) update(v zed.Value) {
-	stash := make(zcode.Bytes, len(v.Bytes))
-	copy(stash, v.Bytes)
-	c.values = append(c.values, zed.Value{v.Type, stash})
-	c.size += len(v.Bytes)
+func (c *Collect) update(val *zed.Value) {
+	stash := make(zcode.Bytes, len(val.Bytes))
+	copy(stash, val.Bytes)
+	c.values = append(c.values, zed.Value{val.Type, stash})
```

Comment:
```suggestion
	c.values = append(c.values, *val.Copy())
```

Reply:
This sends the pointer to GC

Reply:
The compiler's smart enough to put it on the stack.